File: scripts/games/Murray.js
import LeapFrog from "../Leapfrog.js";
import FourThreeTwoOnePoints from "../updateScores/FourThreeTwoOnePoints.js";
import HighestScore from "../endOfGame/HighestScore.js";
import MurrayBoard from "../boards/MurrayBoard.js";

export default class Murray extends LeapFrog {
    constructor() {
        super("Murray");
    }
    initialize(rows, cols) {
        super.initialize(rows, cols);
        this.board = new MurrayBoard().createBoard(rows, cols);
        this.updateScoresStrategy = new FourThreeTwoOnePoints(this);
        this.endOfGameStrategy = new HighestScore(this);
    }
}

File: src/main/java/model/games/Murray.java
package model.games;

import model.LeapFrog;
import model.boards.MurrayBoard;

public class Murray extends LeapFrog {
    public Murray() {
        super("Murray");
    }

    public void initialize(int rows, int cols) {
        super.initialize(rows, cols);
        this.board = new MurrayBoard().createBoard(rows, cols);
    }
}
